src/components/TextareaV2/TextareaV2.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { forwardRef, useCallback, useState } from 'react';
+import cx from 'classnames';
+import { useUID } from 'react-uid';
+import type { TextareaV2Props } from './TextareaV2.types';
+import styles from './TextareaV2.module.css';
+
+const TextareaV2 = forwardRef<HTMLTextAreaElement, TextareaV2Props>(
+  (
+    {
+      classNames,
+      dataTestRef,
+      errorMessage,
+      id: idProp,
+      isEnabled = true,
+      label,
+      maxLength,
+      name,
+      onBlur,
+      onChange,
+      rows,
+      theme = 'dark',
+      value: valueProp,
+    },
+    ref,
+  ) => {
+    const uid = useUID();
+    const textareaId = idProp || uid;
+    const [value, setValue] = useState(valueProp || '');
```

Comment:
Would using nullish coalescing be better? Otherwise a falsy value of `0` would end up being `''`

```suggestion
    const [value, setValue] = useState(valueProp ?? '');
```